engines/scumm/sound.cpp

```diff
@@ -890,7 +890,20 @@ void Sound::startTalkSound(uint32 offset, uint32 b, int mode, Audio::SoundHandle
 			break;
 		default:
 			if (_sampleIsPCMS16BE44100) {
-				offset += 32; // size of VOC header
+				byte *vocHeader = new byte[32];
+
+				file->read(vocHeader, 32);
+				// Don't apply the Indy4 MONSTER.SOU workaround if we don't find the bogus VOC header
+				if (memcmp(vocHeader, "Creative Voice File\x1a\x1a\x00\x0a\x01\x29\x11\x01\x02\x50\x01\xa6\x00", 32) != 0) {
+					_sampleIsPCMS16BE44100 = false;
+				}
+
+				delete[] vocHeader;
+				file->seek(-32, SEEK_CUR);
+			}
+
+			if (_sampleIsPCMS16BE44100) {
```

Comment:
It would be better to merge the code above and this into the same block - there's no reason to re-check this and have a separate block

Reply:
Actually, disregard my comment, it looks better this way :)